packages/integration-sdk-runtime/src/execution/jobState.ts

```diff
@@ -82,6 +82,10 @@ export class MemoryDataStore {
   get(key: string): unknown {
     return this.data[key];
   }
+
+  delete(key: string): void {
+    this.data[key] = undefined;
```

Comment:
I think we should actually delete this key from memory completely instead of assigning the value to `undefined`. 

```js
delete this.data[key];
```